pkg/kubelet/kubelet_pods.go

```diff
@@ -218,11 +218,13 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 		// Docker Volume Mounts fail on Windows if it is not of the form C:/
 		containerPath := mount.MountPath
 		if runtime.GOOS == "windows" {
-			if (strings.HasPrefix(hostPath, "/") || strings.HasPrefix(hostPath, "\\")) && !strings.Contains(hostPath, ":") {
+			// Append C: only if it looks like a local path. Do not process UNC path/SMB shares/named pipes
+			if (strings.HasPrefix(hostPath, "/") || strings.HasPrefix(hostPath, "\\")) && !strings.Contains(hostPath, ":") && !volumeutil.IsWindowsUNCPath(runtime.GOOS, hostPath) {
```

Comment:
what about write as following, it would be more clear:
```
if ((strings.HasPrefix(hostPath, "/") && !strings.Contains(hostPath, ":")) && !volumeutil.IsWindowsUNCPath(runtime.GOOS, hostPath)  {
     hostPath = "c:" + hostPath
}
```

`strings.HasPrefix(hostPath, "\\")` is a redadant check since `IsWindowsUNCPath` also has this check.

Reply:
@andyzhangx the existing `strings.HasPrefix(hostPath, "\\")` is checking for the prefix: `\` as in `\foo\bar`. We should keep that behavior. Note that ```strings.HasPrefix(`\foo\bar`, "\\")``` is `true`

`IsWindowsUNCPath` is checking for prefix `\\` e.g. `\\foo\bar` which is different from what `strings.HasPrefix(hostPath, "\\")` is checking for: `\\` vs `\`

Reply:
> IsWindowsUNCPath is checking for prefix \\ e.g. \\foo\bar which is different from what strings.HasPrefix(hostPath, "\\") is checking for: \\ vs \

 That is very subtle. Add a comment explaining this, so folks are less likely to "fix" this in the future by mistake. Also is there any unit test that can be added to `kubelet_pods_tests.go` for this to catch regressions?

Reply:
Improved the comment to clarify the behavior.